src/System.Windows.Forms/src/System/Windows/Forms/KeysConverter.cs

```diff
@@ -70,45 +65,59 @@ private void Initialize()
             AddKey("7", Keys.D7);
             AddKey("8", Keys.D8);
             AddKey("9", Keys.D9);
+
+            void AddKey(string key, Keys value)
+            {
+                _keyNames[key] = value;
+                _displayOrder!.Add(key);
+            }
         }
 
         /// <summary>
         ///  Access to a lookup table of name/value pairs for keys.  These are localized
         ///  names.
         /// </summary>
-        private IDictionary KeyNames
+        [MemberNotNull(nameof(_keyNames))]
+        [MemberNotNull(nameof(_displayOrder))]
+        private IDictionary<string, Keys> KeyNames
         {
             get
             {
-                if (keyNames is null)
+                if (_keyNames is null)
                 {
-                    Debug.Assert(displayOrder is null);
+                    Debug.Assert(_displayOrder is null);
                     Initialize();
                 }
 
-                return keyNames;
+#pragma warning disable CS8774 // Member must have a non-null value when exiting.
```

Comment:
Apologies, it appears I posted an out-dated suggestion in the last review
```suggestion
#pragma warning disable CS8774 // Member must have a non-null value when exiting: Initialize() inits both _keyNames and _displayOrder.
```